File: Search/FirstBadVersion/FirstBadVersion.java
public class Solution {
	public int firstBadVersion(int ) {
		int start = 0;
		int end = n + 1;
		while(start + 1 < end) {
			int mid = start + (end - start) / 2;
			if(VersionControl.isBadVersion(mid)) {
				end = mid;
			} else {
				start = mid;
			}
		}
		return (start + 1);
	}
}

File: Search/FirstBadVersion/FirstBadVersion.py
class Solution:
    def FirstBadVersion(self, num):
        start, end = 0, num + 1
        while start + 1 < end:
            mid = start + (end - start) / 2
            if VersionControl.isBadVersion(mid):
                end = mid
            else:
                start = mid
               
        return start + 1      
          